qt/scientific_interfaces/test/EnggDiffMultiRunFittingWidgetPresenterTest.h

```diff
@@ -85,6 +86,88 @@ class EnggDiffMultiRunFittingWidgetPresenterTest : public CxxTest::TestSuite {
     assertMocksUsedCorrectly();
   }
 
+  void test_selectRunValidNoFittedPeaks() {
+    auto presenter = setUpPresenter();
+
+    const RunLabel runLabel(123, 1);
+    EXPECT_CALL(*m_mockView, getSelectedRunLabel())
+        .Times(1)
+        .WillOnce(Return(runLabel));
+
+    const boost::optional<Mantid::API::MatrixWorkspace_sptr> sampleWorkspace(
+        WorkspaceCreationHelper::create2DWorkspaceBinned(1, 100));
+    EXPECT_CALL(*m_mockModel, getFocusedRun(runLabel))
+        .Times(1)
+        .WillOnce(Return(sampleWorkspace));
+
+    EXPECT_CALL(*m_mockView, userError(testing::_, testing::_)).Times(0);
+    EXPECT_CALL(*m_mockView, resetCanvas()).Times(1);
```

Comment:
Did you consider splitting this up into multiple separate tests using ON_CALL to fill in for the calls which are only there to provide a return value. This may be why the test has a confusing name?

Reply:
In this case I think it's appropriate to have this all as one test. We're testing the flow of data betwen the model and view, which happens as one conversation, so it seems unnecessary to split it up. I take your point about the calls which are there only to provide a return value, and the repetition in `test_selectValidRunWithFittedPeaks`